### PR TITLE
rgw: Added Cosmetic change in etag of copyObjectResult

### PR DESCRIPTION
**Pull request by shraddhaghatol**
Problem:
Observed discrepancy with AWS for CopyObject API's response.
In ceph-RADOS, "etag" field was not included in  ("/ <etag> /")

Resolution: 
Included etag field of CopyObject API's response in  ("/   /") to make it AWS compliant

Signed-off-by: shraddhaghatol <shraddha.j.ghatol@seagate.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by shraddhaghatol** [comment]
@alimaredia  @cbodley  @dang  Could you please review this PR.

**Review by cdeshmukh — APPROVED** [review]

**Comment by cdeshmukh** [comment]
@cbodley @alimaredia failures seen for 2 checks against this PR doesnt seem to be related to PR changes itself, can you please check & suggest how to correct those

ceph windows tests 
apt-get error: The repository 'https://download.ceph.com/debian-pacific jammy Release' does not have a Release file

make check (arm64)
apt-get error: Malformed line 1 in source list /etc/apt/sources.list.d/libboost.list (type)

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -5517,6 +5517,7 @@ void RGWCopyObj::execute(optional_yield y)
 	   this,
 	   s->yield);
 
+  etag = "\"" + etag + "\"";
```
Comment (cbodley) [inline src/rgw/rgw_op.cc:5520]: we format the response in https://github.com/ceph/ceph/blob/5ab5ff46/src/rgw/rgw_rest_s3.cc#L3470-L3478. please add quotes to the etag like other requests do (ex. https://github.com/ceph/ceph/blob/5ab5ff46/src/rgw/rgw_rest_s3.cc#L1737):

```diff
-      s->formatter->dump_string("ETag", std::move(etag));
+      s->formatter->dump_format("ETag", "\"%s\"", etag.c_str());
```

Comment (shraddhaghatol) [reply src/rgw/rgw_op.cc:5520]: Sure. Will do the change.

**Review by cbodley — APPROVED** [review]

**Comment by shraddhaghatol** [comment]
@cbodley saw your approval. Could you please share the plan to merge this PR.

**Comment by cdeshmukh** [comment]
@cbodley @alimaredia can we merge this please? Am not sure of issues failing in PR checklist & if they apply to this one.

**Comment by cbodley** [comment]
> @cbodley saw your approval. Could you please share the plan to merge this PR.

@shraddhaghatol the `needs-qa` tag means it's ready for testing. we occasionally batch up PRs with this tag and run them through teuthology for validation

if you or @cdeshmukh would like to help test this yourself, the first step is to request vpn access to the sepia lab: https://wiki.sepia.ceph.com/doku.php?id=vpnaccess. when you file a ticket, feel free to link to this PR - Ali and I are both happy to vouch for you

you can read more about the integration tests in https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-intro/#how-to-run-integration-tests. once you have vpn access, we can share the exact command we use to schedule the rgw suite of tests

**Comment by ivancich** [comment]
Since you checked off "references tracker ticket" please provide a link to that ticket in the description, @shraddhaghatol .

**Comment by shraddhaghatol** [comment]
> Since you checked off "references tracker ticket" please provide a link to that ticket in the description, @shraddhaghatol .

@ivancich  Added this check accidently. Removed it now.

**Comment by cbodley** [comment]
@shraddhaghatol @cdeshmukh i've added this to my wip-cbodley-testing branch with 3 other PRs. i pushed that branch to the ceph-ci repo at https://github.com/ceph/ceph-ci/commits/wip-cbodley-testing, which triggers CI to build packages/containers that will show up under https://shaman.ceph.com/builds/ceph/wip-cbodley-testing/. once the builds are done, i'll schedule the rgw suite with a command like this:

> $ teuthology-suite -s rgw -m smithi -c wip-cbodley-testing --subset 1/5 -p 75

test results will eventually show up under https://pulpito.ceph.com/?suite=rgw. i'll follow up with a link to the results, and merge if there are no regressions

**Comment by cbodley** [comment]
jenkins test api

**Comment by cdeshmukh** [comment]
> if you or @cdeshmukh would like to help test this yourself, the first step is to request vpn access to the sepia lab:

thanks @cbodley for the suggestion, let us get acquainted with teuthology first and may be in sometime future we can plan for PRs coming from [cortx-rgw](https://github.com/Seagate/cortx-rgw), so not for this PR at least.

**Comment by cbodley** [comment]
> thanks @cbodley for the suggestion, let us get acquainted with teuthology first and may be in sometime future we can plan for PRs coming from [cortx-rgw](https://github.com/Seagate/cortx-rgw), so not for this PR at least.

sounds good! we welcome your contributions either way :heart:

**Comment by cdeshmukh** [comment]
thanks @cbodley

**Pull request merged**